doc/source/methods/modeldistillation.ipynb

```diff
@@ -0,0 +1,180 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "[source](../api/alibi_detect.ad.model_distillation.rst)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Model distillation adversarial and drift detector"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Overview\n",
+    "\n",
+    "[Model distillation](https://arxiv.org/abs/1503.02531) is a technique that is used to transfer knowledge from a large network to a smaller network. Tipically, it consists of training an ancillary model with a simplified architecture on soft targets (the output distributions or the logits) obtained from the original model. \n",
+    "\n",
+    "Here, we apply model distillation to obtain harmfulness scores, by comparing the output distributions of the original model with the output distributions \n",
+    "of the distilled model, in order to detect adversarial data, malicious data drift or data corruption.\n",
+    "We use the following definition of harmful and harmless data points:\n",
+    "* Harmful data points are defined as inputs for which the model's predictions on the uncorrupted data are correct while the model's predictions on the corrupted data are wrong.\n",
+    "* Harmless data points are defined as inputs for which the model's predictions on the uncorrupted data are correct and the model's predictions on the corrupted data remain correct.\n",
+    "\n",
+    "Analogously to the [adversarial detector](https://arxiv.org/abs/2002.09364), which is also part of the library, the model distillation detector picks up drift that reduces the performance of the classification model. \n",
+    "\n",
+    "The detector can be used as follows:\n",
+    "\n",
+    "* An adversarial score $S$ is computed. $S$ equals the loss function employed for distillation between the model predictions on $x$ and $x'$.\n",
+    "\n",
+    "* If $S$ is above a threshold (explicitly defined or inferred from training data), the instance is flagged as adversarial."
```

Comment:
I find the paragraph a bit confusing because it jumps back and forth between adversarial and drift detection. I think it would be helpful to emphasise that the detector component is an adversarial detector, and explain that it computes the adversarial score by computing a divergence metric like the K-L between the prediction distributions of the original and distilled model. If those prediction distributions diverge and the adversarial score is above a threshold, the instance is flagged as adversarial.

However, the adversarial scores can also be used in the context of drift detection. By computing e.g. a K-S or MMD statistic between the adversarial scores on a batch of reference data and a batch of incoming test data, malicious data drift between both batches can be detected. Then we can add the definition of "malicious", and also mention that this is very similar to how the Adversarial Auto-Encoder detector is used for drift detection and provide links to the relevant notebooks.